modules/oe_content_publication/oe_content_publication.post_update.php

```diff
@@ -34,41 +35,87 @@ function oe_content_publication_post_update_00002() {
+    'composite_reference',
+    'entity_reference_revisions',
+    'field_group',
+    'inline_entity_form',
+    'link',
+    'path',
+    'oe_content',
+    'oe_content_departments_field',
+    'oe_content_documents_field',
+    'oe_content_entity_contact',
+    'oe_content_organisation_reference',
+    'oe_content_reference_code_field',
+    'oe_media_avportal',
+  ];
+
+  foreach ($modules as $module) {
+    if (!$module_handler->moduleExists($module)) {
+      $module_installer->install([$module]);
+    }
+  }
 
   // Clear the cached plugin definitions of the field types.
   \Drupal::service('plugin.manager.field.field_type')->clearCachedDefinitions();
+}
 
-  // Create the field storage for the reference field.
-  $field_storage_config = \Drupal::service('entity_type.manager')->getStorage('field_storage_config');
-  if (!$field_storage_config->load('node.oe_publication_contacts')) {
-    $reference_field = $storage->read('field.storage.node.oe_publication_contacts');
-    // We are creating the config which means that we are also shipping
-    // it in the config/install folder so we want to make sure it gets the hash
-    // so Drupal treats it as a shipped config. This means that it gets exposed
-    // to be translated via the locale system as well.
-    $reference_field['_core']['default_config_hash'] = Crypt::hashBase64(serialize($reference_field));
-    $field_storage_config->create($reference_field)->save();
+/**
+ * Create the new fields in the Publication content type.
+ */
+function oe_content_publication_post_update_00004(): void {
+  $storage = new FileStorage(drupal_get_path('module', 'oe_content_publication') . '/config/post_updates/00004_create_fields');
+  \Drupal::service('config.installer')->installOptionalConfig($storage);
+}
+
+/**
+ * Update field settings.
+ */
+function oe_content_publication_post_update_00005(): void {
+  $new_field_labels = [
+    'node.oe_publication.oe_publication_type' => 'Resource type',
+    'node.oe_publication.oe_documents' => 'Files',
+  ];
+  foreach ($new_field_labels as $id => $new_label) {
+    $field_config = FieldConfig::load($id);
+    $field_config->setLabel($new_label);
+    $field_config->save();
   }
 
-  // Create the field config for the reference field.
-  $field_config = \Drupal::service('entity_type.manager')->getStorage('field_config');
-  if (!$field_config->load('node.oe_publication.oe_publication_contacts')) {
-    $reference_field = $storage->read('field.field.node.oe_publication.oe_publication_contacts');
-    $reference_field['_core']['default_config_hash'] = Crypt::hashBase64(serialize($reference_field));
-    $field_config->create($reference_field)->save();
+  $new_translatable_settings = [
+    'node.oe_publication.oe_summary' => FALSE,
+    'node.oe_publication.oe_teaser' => FALSE,
+  ];
+  foreach ($new_translatable_settings as $id => $value) {
+    $field_config = FieldConfig::load($id);
+    $field_config->setTranslatable($value);
+    $field_config->save();
```

Comment:
Let's not update the field translatablility for existing fields, we have a dedicated issue for that: EWPP-169.

Reply:
The code which updates the translatability is removed.